# Projects

## 1. Viral infection global analyses:
* Transcriptome analyses to select differentially expressed genes
* Isoform differential expression
* Map RNAseq to virus
* Functional and enrichment analyses
* Search for global modulators of disease virulence and host susceptibility

## 2. Human-virus interaction analyses:
* Which human mRNAs and proteins interact with or are regulating the virus and vice-versa?
* Are there human RNA-binding proteins potentially regulating the viral genome?
* Include these in subsequent analyses

## 3. Increased risk factors analyses:
* Select datasets from co-morbidities related to severeness of Covid-19
For instance, diseases such as diabetes myelitus and hypertension; other factors like smoking, which might make the person more vulnerable to the virus;
* Select other human tissues to check the expression of proteins interacting with the virus (selected in previous step)
* Search for SNPs, splicing variants, regulatory regions for all genes selected in previous steps
* Analysis of HLA types that predispose individuals and populations to COVID-19 infection and mortality (starting at http://hlacovid19.org/
* Analysis of host expression differences in ACE2, TMPRSS2, and other key genes involved in SARS-CoV-2 infection. See ACE2 expression in normal lung from GTExv8 here: https://genenetwork.org/show_trait?trait_id=ENSG00000130234&dataset=GTEXv8_Lung_tpm_0220
* Analysis of BXD mouse models in viral pneumonia susceptibility after viral infection. See Ace2 expression for 43 genomes here: https://genenetwork.org/show_trait?trait_id=ENSMUSG00000015405&dataset=HZI_LungBXD_RNA-Seq_1116


## 4. Towards the control and sub typing of gene expression:

* Are there any known drugs or other factors that might modulate the expression of selected genes? How do drugs affect expression in different backgrounds?   

In this section we can also communicate with #htvs (Virtual Screening) as they “will use target binding of the crystal structure of the viral protease with potential inhibitors”, so maybe there is room for collaborating their pipeline on human proteins as well? We can also collaborate with #annotations on structures.  

Initial work: 
https://amp.pharm.mssm.edu/covid19/
https://www.nature.com/articles/s41467-019-08831-9
https://science.sciencemag.org/content/367/6473/45.full
https://www.biorxiv.org/content/10.1101/2020.03.22.002386v1
* https://www.covid19hg.org/
* https://www.covidhge.com/

Comparing drug response RNAseq profiles, as well as making it possible to detect chronic diseases that may arise from CoVID-19. https://github.com/NCBI-Codeathons/ViraVate

## 5. Reporting findings to electronic medical records: 
* How do we get RNAseq findings into the hands of clinicians in both the immediate and longer (chronic disease) term?  

Initial work: 
https://github.com/NCBI-Codeathons/omopomics -- working with OHDSI

