# Project 1: SARS-CoV-2 infection global analyses

## Software
1. Alignment
    - STAR: v2.5.2b
    - samtools: v1.9 (using htslib 1.9)
2. Read counts
    - StringTie: v2.1.1
    - gffcompare

## References
1. Homo sapiens
    - GRCh38: Gencode 24

## File structure
1. fastq (input data)
2. bam (results generated by STAR)
3. counts (results from StringTie)
    - final (created during execution)
4. genomes (reference genomes)
    - GRCh38 (Homo sapiens build 38)
        - starIndex_gencode24 (Gencode version ready for STAR)
5. scripts (to be run in order)
    - alignment.sh
    - counts_p1.sh
    - counts_p2.sh
    - counts_p3.sh
    - DE_prep.sh
    - prepDE.py (auxiliary, from the creators of StringTie)
