*Software*

1) Alignment
- STAR: v2.5.2b
- samtools: v1.9 (using htslib 1.9)

2) Read counts
- StringTie: v2.1.1

*References*
1) Homo sapiens
- GRCh38: Gencode 24

*File structure*
.
|-- bam (results generated by STAR)
|-- counts (results from StringTie)
|   |-- final (created during execution)
|-- fastq (input data)
|-- genomes (reference genome ready for STAR)
|   `-- GRCh38
|       `-- starIndex_gencode24
`-- scripts (to be run in order)
    |-- alignment.sh
    |-- counts_p1.sh
    |-- counts_p2.sh
    |-- counts_p3.sh
    |-- DE_prep.sh
    `-- prepDE.py (auxiliary, from the creators of StringTie)
