# Project 1: SARS-CoV-2 infection global analyses

## Software
1. Alignment
    - STAR: v2.5.2b
    - samtools: v1.9 (using htslib 1.9)
2. Read counts
    - StringTie: v2.1.1
    - gffcompare

## References
1. Homo sapiens
    - GRCh38: Gencode 24

## Directory structure
1. fastq (input data)
2. bam (results generated by STAR)
3. counts (results from StringTie)
    - final (created during execution)
4. genomes (reference genomes)
    - GRCh38 (Homo sapiens build 38)
        - starIndex_gencode24 (Gencode version ready for STAR)
5. scripts (to be run in order)
    - alignment.sh
    - counts_p1.sh
    - counts_p2.sh
    - counts_p3.sh
    - DE_prep.sh
    - prepDE.py (auxiliary, from the creators of StringTie)

## Software used in this project
### Command line tools
- FastQC (https://github.com/s-andrews/FastQC) [docker image](quay.io/biocontainers/fastqc:0.11.5--1)
- Fastq-screen (https://www.bioinformatics.babraham.ac.uk/projects/fastq_screen/) [docker image](quay.io/biocontainers/fastq-screen:0.11.3--pl5.22.0_0)
- trimmomatic (http://www.usadellab.org/cms/?page=trimmomatic)[docker image](quay.io/biocontainers/trimmomatic:0.33--1)
- STAR (https://github.com/alexdobin/STAR)[docker image](quay.io/biocontainers/star:2.7.3a--0)
- Bowtie2 (http://bowtie-bio.sourceforge.net/bowtie2/index.shtml)[docker image](quay.io/biocontainers/bowtie2:2.4.1--py38he513fc3_0)
- HISAT2 (https://daehwankimlab.github.io/hisat2/)[docker image](quay.io/biocontainers/hisat2:2.2.0--py37he1b5a44_0)
- TEtools (https://github.com/l-modolo/TEtools)[docker image](zavolab/tetools:1.0.0) **NB:** the command passed to Docker must specify python and the full path to the toolname, like so: `python3 /TEtools/TEcount.py`.
- TEtranscripts (http://hammelllab.labsites.cshl.edu/software)[docker image](quay.io/biocontainers/tetranscripts:2.1.4--py_0)
- LIONS (www.github.com/ababaian/LIONS)[docker image](hpobiolab/lions)
- samtools / picard (http://samtools.sourceforge.net/; https://broadinstitute.github.io/picard/)[docker image](quay.io/biocontainers/picard:2.22.2--0)
- featureCounts (http://subread.sourceforge.net)[docker image](hpobiolab/feature-counts)
- MultiQC (https://github.com/ewels/MultiQC)[docker image](quay.io/biocontainers/multiqc:1.8--py_2)
- KissSplice (http://kissplice.prabi.fr/)[docker image](hpobiolab/kiss-splice)

### R packages

Docker image: `hpobiolab/gene-expression-r`

#### Direct install
   _ggplot2_
```
install.packages("ggplot2")
```

#### BiocManager install
```
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
```
_DESEq2_
```
BiocManager::install("DESeq2")
```
_EdgeR_
```
BiocManager::install("EdgeR")
```
_Limma-Voom_
```
BiocManager::install("limma")
```

#### Devtools install
```
install.packages("devtools")
```
_SARTools_
```
library(devtools)
install_github("PF2-pasteur-fr/SARTools", build_opts="--no-resave-data")
```

## Virus infection studies
- SARS-MERS: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE56192  
- COVID: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE147507 
- Murine coronavirus (M-CoV): https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-4111/

