# Project 1: Global Host Antiviral Response
    -A separate workflow to enable future comparisons of results.

## Software
1. Data acquisition:
    - NCBI sratools
    
    The following steps were performed with Automated Reproducible MOdular Workflow for Preprocessing and Differential Analysis of RNA-seq Data(ARMOR; https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6643886/)
2. Preprocessing Workflow
    A. Read trimming (cutadapt; DOI:10.14806/ej.17.1.200)
    B. Read mapping and quantification (salmon; https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5600148/)
    C. Differential Expression (edgeR; https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2796818/)
    D. Functional Enrichment (camera; https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3458527/)
    E. Isoform analysis (DRIMseq; https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5200948/)
3. Postprocessing:
    A. Signaling pathway impact analysis (SPIA; https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2732297/)

## References
1. Homo sapiens
    - GRCh38: Ensembl 98

## Directory structure
1. fastq (input data) from SRA
2. trimmed fastq in VM 
    - stored in /home/shared/raw/FASTQtrimmmed/
3. normalized counts (results from salmon)
    - stored in /home/shared/analyzed/salmon_quant
4. Reference genomes (reference genomes)
    - stored in /home/shared/raw/Homo_sapiens.GRCh38.98.gtf
5. Output
    - all_results: contains all files generated by the ARMOR workflow for each of the three viral datasets
        -/home/shared/analyzed/all_results
    - summary_results: contains the files from the DE, isoform, functional enrichment, and pathway analyses for each of the three datasets
        -/home/shared/analyzed/summary_results
    - Combined_Functions.txt: contains a matrix of all unique functional terms and each experimental comparison. This enables the identification of shared/unique terms associated with infection by each virus
        -/home/shared/analyzed/
    - Combined_Pathways.txt: contains a matrix of all unique signaling pathways and each experimental comparison. This enables the identification of shared/unique pathways associated with infection by each virus
        - /home/shared/analyzed/
6. Metadata: A file containing the SRR IDs from the sequence read archive at NCBI, experimental variable(s) represented by each ID, and the sequencing library type (single- or paired-end)

## Software used in this project
1. ARMOR (https://github.com/csoneson/ARMOR)
2. cutadapt (https://github.com/marcelm/cutadapt)
3. salmon (https://salmon.readthedocs.io/en/latest/salmon.html)
4. edgeR (https://bioconductor.org/packages/release/bioc/html/edgeR.html)
5. camera, in limma package (https://bioconductor.org/packages/release/bioc/html/limma.html)
6. DRIMseq (http://www.bioconductor.org/packages/release/bioc/html/DRIMSeq.html)
7. SPIA (http://www.bioconductor.org/packages/release/bioc/html/SPIA.html)

## Virus infection studies
- SARS-MERS: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE56192  
- SARS-CoV-2: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE147507 
- MERS: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE122876

